src/NosCore.Controllers/Miniland/MinilandObjects/MgPacketHandler.cs

```diff
@@ -43,137 +43,136 @@ public class MgPacketHandler : PacketHandler<MinigamePacket>, IWorldPacketHandle
     {
         private readonly IItemProvider _itemProvider;
         private readonly IMinilandProvider _minilandProvider;
-        private ClientSession _clientSession;
-        private MinigamePacket _minigamePacket;
-        private GameObject.Providers.MinilandProvider.Miniland _miniland;
-        private MapDesignObject _minilandObject;
+        private ClientSession? _clientSession;
+        private MinigamePacket? _minigamePacket;
+        private GameObject.Providers.MinilandProvider.Miniland? _miniland;
+        private MapDesignObject? _minilandObject;
 
         public MgPacketHandler(IMinilandProvider minilandProvider, IItemProvider itemProvider)
         {
             _minilandProvider = minilandProvider;
             _itemProvider = itemProvider;
         }
 
-        public override Task Execute(MinigamePacket minigamePacket, ClientSession clientSession)
+        public override async Task Execute(MinigamePacket minigamePacket, ClientSession clientSession)
         {
             _clientSession = clientSession;
             _minigamePacket = minigamePacket;
             _miniland = _minilandProvider.GetMiniland(clientSession.Character.CharacterId);
             _minilandObject =
-                clientSession.Character.MapInstance.MapDesignObjects.Values.FirstOrDefault(s =>
+                clientSession.Character.MapInstance!.MapDesignObjects.Values.FirstOrDefault(s =>
                     s.Slot == minigamePacket.Id);
             if ((_minilandObject == null) || (_miniland == null))
             {
-                return Task.CompletedTask;
+                return;
             }
 
-            if (_minilandObject.InventoryItemInstance.ItemInstance.Item.IsWarehouse)
+            if (_minilandObject?.InventoryItemInstance?.ItemInstance?.Item?.IsWarehouse != false)
             {
-                return Task.CompletedTask;
+                return;
             }
 
-            var game = (byte) (_minilandObject.InventoryItemInstance.ItemInstance.Item.EquipmentSlot ==
+            var game = (byte)(_minilandObject.InventoryItemInstance.ItemInstance.Item.EquipmentSlot ==
                 EquipmentType.MainWeapon
                     ? (4 + _minilandObject.InventoryItemInstance.ItemInstance.ItemVNum) % 10
-                    : (int) _minilandObject.InventoryItemInstance.ItemInstance.Item.EquipmentSlot / 3);
+                    : (int)_minilandObject.InventoryItemInstance.ItemInstance.Item.EquipmentSlot / 3);
             var full = false;
 
             switch (minigamePacket.Type)
             {
                 case 1:
-                    Play(game);
+                    await Play(game);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Add '.ConfigureAwait(false)' to this call to allow execution to continue in any thread.](https://app.codacy.com/manual/NosCoreIO/NosCore/pullRequest?prid=5262615)